modules/xfem/src/userobjects/GeometricCutUserObject.C

```diff
@@ -34,14 +37,25 @@ validParams<GeometricCutUserObject>()
 }
 
 GeometricCutUserObject::GeometricCutUserObject(const InputParameters & parameters)
-  : CrackFrontPointsProvider(parameters)
+  : CrackFrontPointsProvider(parameters), _heal_mesh(getParam<bool>("heal_mesh"))
 {
-  FEProblemBase * fe_problem = dynamic_cast<FEProblemBase *>(&_subproblem);
-  if (fe_problem == NULL)
-    mooseError("Problem casting _subproblem to FEProblemBase in XFEMMaterialStateMarkerBase");
-  _xfem = MooseSharedNamespace::dynamic_pointer_cast<XFEM>(fe_problem->getXFEM());
+  _xfem = MooseSharedNamespace::dynamic_pointer_cast<XFEM>(_fe_problem.getXFEM());
   if (_xfem == NULL)
-    mooseError("Problem casting to XFEM in XFEMMaterialStateMarkerBase");
+    mooseError("Problem casting to XFEM in GeometricCutUserObject");
+
+  _xfem->addGeometricCut(this);
+
+  MooseSharedPointer<XFEMElementPairLocator> new_xfem_epl(
+      new XFEMElementPairLocator(_xfem, _interface_id));
```

Comment:
Please don't use MooseSharedPointer, it's deprecated (use std::shared_pointer). Also, use `std::make_shared()` instead of calling `new`.

Reply:
I will make the changes to other files too.